subsys/usb/device/class/dfu/usb_dfu.c

```diff
@@ -655,9 +684,19 @@ static int dfu_class_handle_to_device(struct usb_setup_packet *setup,
 
 		/* Move to appDETACH state */
 		dfu_data.state = appDETACH;
+#if IS_ENABLED(CONFIG_USB_DFU_WILL_DETACH)
+		/* FIXME: Detach should happen once the status stage finishes
```

Comment:
You can replace `FIXME:` with `Note:` . 
Please also replace #if directives with if statements, like:
```diff c
diff --git a/subsys/usb/device/class/dfu/usb_dfu.c b/subsys/usb/device/class/dfu/usb_dfu.c
index 05cb61a5423..c51e36c835f 100644
--- a/subsys/usb/device/class/dfu/usb_dfu.c
+++ b/subsys/usb/device/class/dfu/usb_dfu.c
@@ -684,19 +684,19 @@ static int dfu_class_handle_to_device(struct usb_setup_packet *setup,
 
                /* Move to appDETACH state */
                dfu_data.state = appDETACH;
-#if IS_ENABLED(CONFIG_USB_DFU_WILL_DETACH)
-               /* FIXME: Detach should happen once the status stage finishes
-                * but the USB device stack does not expose such callback.
-                * Wait fixed time (ignore wValue) to let device finish control
-                * transfer status stage.
-                */
-               timeout = CONFIG_USB_DFU_DETACH_TIMEOUT;
-               k_timer_start(&dfu_timer, K_MSEC(timeout), K_FOREVER);
-#else
-               /* Begin detach timeout timer */
-               timeout = MIN(setup->wValue, CONFIG_USB_DFU_DETACH_TIMEOUT);
+               if (IS_ENABLED(CONFIG_USB_DFU_WILL_DETACH)) {
+                       /* Note: Detach should happen once the status stage
+                        * finishes but the USB device stack does not expose
+                        * such callback. Wait fixed time (ignore wValue) to
+                        * let device finish control transfer status stage.
+                        */
+                       timeout = CONFIG_USB_DFU_DETACH_TIMEOUT;
+               } else {
+                       /* Begin detach timeout timer */
+                       timeout = MIN(setup->wValue, CONFIG_USB_DFU_DETACH_TIMEOUT);
+               }
+
                k_timer_start(&dfu_timer, K_MSEC(timeout), K_FOREVER);
-#endif
                break;
        default:
                LOG_DBG("Unsupported bmRequestType 0x%02x bRequest 0x%02x",
```

Reply:
The `if` instead of `#ifdef` is quite interesting mechanism. The way I understand it is that it is assumed that the compiler performs dead code removal (so the resulting binary is same size as with `#ifdef`) and the advantage is that all code paths do need to contain code that compiles (so there's no risk that the other path won't even compile as the development goes on).

Unfortunately the increased indentation makes the `timeout = MIN(setup->wValue, CONFIG_USB_DFU_DETACH_TIMEOUT);` get past 80 characters with tab size 8.